File: examples/fp-verification-demo-python/src/tp_true_positives.py
# TP — True Positives: 真实漏洞，必须检出
import os
import subprocess


def vulnerable_query(conn, user_id):
    """TP-01: SQL 注入 — f-string 拼接"""
    cursor = conn.cursor()
    cursor.execute(f"SELECT * FROM users WHERE id = {user_id}")  # 注入！


def vulnerable_exec(input_cmd):
    """TP-02: 命令注入 — os.system 未过滤"""
    os.system(f"ping {input_cmd}")  # 注入！

File: examples/fp-verification-demo-java/src/tp_true_positives.java
// TP — True Positives: 真实漏洞，必须检出
import java.sql.*;

public class tp_true_positives {

    // TP-01: SQL 注入 — 字符串拼接
    public void vulnerableQuery(Connection conn, String userId) throws SQLException {
        String sql = "SELECT * FROM users WHERE id = " + userId;  // 拼接！
        try (Statement stmt = conn.createStatement()) {
            ResultSet rs = stmt.executeQuery(sql);
        }
    }

    // TP-02: 命令注入 — Runtime.exec 未过滤
    public void vulnerableExec(String input) throws Exception {
        String cmd = "ping " + input;  // 拼接！
        Runtime.getRuntime().exec(cmd);
    }
}
